### ISSUE TITLE
ChatBedrock: I am unable to stream when using with_structured_output. I can either stream or I can use with_structured_output.

### ISSUE THREAD

**Issue by OSS-GR** [langchain-ai/langchain#25056]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

``` python 
class CitationSubModel(TypedDict):
    number: int = Field(description="An integer numbering the citation i.e. 1st citation, 2nd citation.")
    id: str = Field(description="The identifiying document name or number for a document.")

class FinalAnswerModel(TypedDict):
    answer: Annotated[str, ..., "The answer to the user question using the citations"]
    citations: Annotated[List[CitationSubModel], ..., "A dictionary that includes the numbering and the id references for the citations to be used to answer"]

model_answer = get_model(state.get("deployment_id","gpt-4o-global"), streaming=True)
model_answer = model_answer.with_structured_output(FinalAnswerModel)
# Create chain using chat_history, prompt_template and the model. Parse results through a simple string parser.
chain = (RunnablePassthrough.assign(chat_history=RunnableLambda(memory.load_memory_variables) | itemgetter("chat_history")) | prompt_template | model_answer)

for chunk in chain.stream({"context": context, "task": state["task"], "citation": citation_instruction, "coding_instructions": coding_instructions}):
    print(chunk)    
```

### Error Message and Stack Trace (if applicable)

``` python
File "/var/task/chatbot_workflow.py", line 858, in solve
    for chunk in chain.stream({"context": context, "task": state["task"], "citation": citation_instruction, "coding_instructions": coding_instructions}):
  File "/opt/python/lib/python3.10/site-packages/langchain_core/runnables/base.py", line 3253, in stream
    yield from self.transform(iter([input]), config, **kwargs)
  File "/opt/python/lib/python3.10/site-packages/langchain_core/runnables/base.py", line 3240, in transform
    yield from self._transform_stream_with_config(
  File "/opt/python/lib/python3.10/site-packages/langchain_core/runnables/base.py", line 2053, in _transform_stream_with_config
    chunk: Output = context.run(next, iterator)  # type: ignore
  File "/opt/python/lib/python3.10/site-packages/langchain_core/runnables/base.py", line 3202, in _transform
    for output in final_pipeline:
  File "/opt/python/lib/python3.10/site-packages/langchain_core/runnables/base.py", line 1271, in transform
    for ichunk in input:
  File "/opt/python/lib/python3.10/site-packages/langchain_core/runnables/base.py", line 5267, in transform
    yield from self.bound.transform(
  File "/opt/python/lib/python3.10/site-packages/langchain_core/runnables/base.py", line 1289, in transform
    yield from self.stream(final, config, **kwargs)
  File "/opt/python/lib/python3.10/site-packages/langchain_core/language_models/chat_models.py", line 373, in stream
    raise e
  File "/opt/python/lib/python3.10/site-packages/langchain_core/language_models/chat_models.py", line 353, in stream
    for chunk in self._stream(messages, stop=stop, **kwargs):
  File "/opt/python/lib/python3.10/site-packages/langchain_aws/chat_models/bedrock.py", line 439, in _stream
    result = self._generate(
  File "/opt/python/lib/python3.10/site-packages/langchain_aws/chat_models/bedrock.py", line 518, in _generate
    for chunk in self._stream(messages, stop, run_manager, **kwargs):
  File "/opt/python/lib/python3.10/site-packages/langchain_aws/chat_models/bedrock.py", line 439, in _stream
    result = self._generate(
  File "/opt/python/lib/python3.10/site-packages/langchain_aws/chat_models/bedrock.py", line 518, in _generate
    for chunk in self._stream(messages, stop, run_manager, **kwargs):
  File "/opt/python/lib/python3.10/site-packages/langchain_aws/chat_models/bedrock.py", line 439, in _stream


### .
### .           Error Message Repeats many times
### .


  File "/opt/python/lib/python3.10/site-packages/langchain_aws/chat_models/bedrock.py", line 439, in _stream
    result = self._generate(
  File "/opt/python/lib/python3.10/site-packages/langchain_aws/chat_models/bedrock.py", line 518, in _generate
    for chunk in self._stream(messages, stop, run_manager, **kwargs):
  File "/opt/python/lib/python3.10/site-packages/langchain_aws/chat_models/bedrock.py", line 439, in _stream
    result = self._generate(
  File "/opt/python/lib/python3.10/site-packages/langchain_aws/chat_models/bedrock.py", line 514, in _generate
    self._get_provider(), "stop_reason"
  File "/opt/python/lib/python3.10/site-packages/langchain_aws/llms/bedrock.py", line 594, in _get_provider
    if self.model_id.startswith("arn"):
RecursionError: maximum recursion depth exceeded while calling a Python object
Stack (most recent call last):
  File "/var/lang/lib/python3.10/threading.py", line 973, in _bootstrap
    self._bootstrap_inner()
  File "/var/lang/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/var/lang/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/var/lang/lib/python3.10/concurrent/futures/thread.py", line 83, in _worker
    work_item.run()
  File "/var/lang/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/opt/python/lib/python3.10/site-packages/langchain_core/runnables/config.py", line 587, in wrapper
    return func(*args, **kwargs)
  File "/var/task/chatbot_workflow.py", line 204, in wrap_func
    result = func(*args, **kwargs)
  File "/var/task/chatbot_workflow.py", line 917, in solve
    logging.exception(f"Error Streaming Chunk: {e}", exc_info=True, stack_info=True)
```

### Description

I have a custom function that calls get_model that returns either an chatazureopenai or chatbedrock object. I then pass the pydantic object to the chat object and stream the response. This works perfectly fine with ChatAzureOpenai but it fails with ChatBedrock.

I am trying to stream the response in a structured output. I have created a TypedDict pydantic class and I am trying to stream as it is generating.

I am getting the following error in a loop as my code is using langgraph and it starts iterating until it hits the max recursion limit.

Not sure what is causing the issue from this error trace. Can anyone help?

### System Info

System Information
------------------
AWS Lambda ARM
Python Version:  3.10

Package Information
-------------------
langchain_core: 0.2.27
langchain: 0.2.11
langchain_community: 0.2.5
langchain_aws: 0.1.13
langchain_openai: 0.1.20
langchainhub: 0.1.14
langgraph: 0.1.19

**Comment by OSS-GR** [comment]
Even a simpler example taken from the chatbedrock documentation and altered to use streaming leads to the same error trace:

``` python
from langchain_aws.chat_models.bedrock import ChatBedrock
import logging

logging.basicConfig(level=logging.DEBUG)

schema = {
    "name": "AnswerWithJustification",
    "description": "An answer to the user question along with justification for the answer.",
    "input_schema": {
        "type": "object",
        "properties": {
            "answer": {"type": "string"},
            "justification": {"type": "string"},
        },
        "required": ["answer", "justification"]
    }
}


llm =ChatBedrock(
    model_id="anthropic.claude-3-sonnet-20240229-v1:0",
    model_kwargs={"temperature": 0.001},
    region_name="us-east-1",
    streaming=True
)  # type: ignore[call-arg]
structured_llm = llm.with_structured_output(schema)

for chunk in structured_llm.stream("What weighs more a pound of bricks or a pound of feathers"):

    print(chunk)

```

**Comment by ccurme** [comment]
Thanks for reporting this. I see you're on `langchain-aws==0.1.13`. A fix for this was recently released. Please `pip install --upgrade langchain-aws` and let me know if the problem persists. Closing for now but will re-open if needed.

**Comment by OSS-GR** [comment]
Hi there, I upgraded to the latest version 0.1.15 and now I do not see the above error message. I am facing a different bug now. Although I am using a for loop with the following logic:
``` python
for chunk in model_with_structured_output.stream({various_inputs: "abv"}):
    print("chunk")

```
the response is not streaming but I simply get the final object once it is generated. In essence it is operating as if I have used .invoke instead of .stream which is not the case. 

In addition, not sure if this is related I am hitting a lot of throttling exceptions which I never have before.
